### PR TITLE
nits for add support for map into reflectcodec 

### PR DESCRIPTION
**Pull request by danlaine**
Some more nits on #1790 

## Why this should be merged

See my comments on this PR

## How this works

Nits

## How this was tested

Existing UT

### CODE REVIEW

File: codec/reflectcodec/type_codec.go [hunk 1]
```diff
@@ -363,26 +363,30 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
...
 			p := wrappers.Packer{
 				MaxSize: int(c.maxSliceLen),
 				Bytes:   make([]byte, 0, 128),
 			}
 			if err := c.marshal(key, &p, c.maxSliceLen); err != nil {
 				return err
 			}
+			if p.Err != nil {
+				return fmt.Errorf("couldn't marshal map key %+v: %w ", key, p.Err)
+			}
```
Comment (danlaine) [inline codec/reflectcodec/type_codec.go:389]: Need to check error here

File: codec/reflectcodec/type_codec.go [hunk 2]
```diff
@@ -394,13 +398,13 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 		})
 
 		for _, key := range sortedKeys {
-			// serialize key
+			// pack key
```
Comment (danlaine) [inline codec/reflectcodec/type_codec.go:401]: Technically, we serialized it above

File: codec/reflectcodec/type_codec.go [hunk 3]
```diff
@@ -607,44 +611,46 @@ func (c *genericCodec) unmarshal(p *wrappers.Packer, value reflect.Value, maxSli
...
-		var lastKey []byte
+		// Set [value] to be a new map of the appropriate type.
+		value.Set(reflect.MakeMapWithSize(mapType, numElts))
 
 		for i := 0; i < numElts; i++ {
 			mapKey := reflect.New(mapKeyType).Elem()
-			mapValue := reflect.New(mapValueType).Elem()
 
-			keyStartPosition := p.Offset
+			keyStartOffset := p.Offset
```
Comment (danlaine) [inline codec/reflectcodec/type_codec.go:628]: Thought this name was maybe a bit more clear

File: codec/reflectcodec/type_codec.go [hunk 4]
```diff
@@ -607,44 +611,46 @@ func (c *genericCodec) unmarshal(p *wrappers.Packer, value reflect.Value, maxSli
...
-			// the serialization spec is to sort map keys before serializing,
-			// for consistency.
+			// We do this to enforce that key-value pairs are sorted by increasing key.
 			//
 			// See for context: https://github.com/ava-labs/avalanchego/pull/1790#discussion_r1283656558
-			keyBytes := p.Bytes[keyStartPosition:p.Offset]
-			if lastKey != nil && bytes.Compare(keyBytes, lastKey) < 0 {
-				return fmt.Errorf("keys are in the incorrect order (%s, %s)", lastKey, mapKey)
+			keyBytes := p.Bytes[keyStartOffset:p.Offset]
+			if i != 0 && bytes.Compare(keyBytes, prevKey) <= 0 {
```
Comment (danlaine) [inline codec/reflectcodec/type_codec.go:642]: I think checking for `i != 0` (i.e. this isn't the first iteration) is a little more readable than checking for `lastKey != nil`

Comment (danlaine) [reply codec/reflectcodec/type_codec.go:642]: Also, we want to enforce that `keyBytes` is _strictly larger_ than `prevKey` (i.e. we need <=, not <`

File: codec/reflectcodec/type_codec.go [hunk 5]
```diff
@@ -607,44 +611,46 @@ func (c *genericCodec) unmarshal(p *wrappers.Packer, value reflect.Value, maxSli
...
+				return fmt.Errorf("keys aren't sorted: (%s, %s)", prevKey, mapKey)
 			}
-			lastKey = keyBytes
+			prevKey = keyBytes
 
+			// Get the value
+			mapValue := reflect.New(mapValueType).Elem()
```
Comment (danlaine) [inline codec/reflectcodec/type_codec.go:648]: Moved declaration closer to use

**Review by ceyonur — APPROVED** [review]

**Review by nytzuga — APPROVED** [review]

**Pull request merged**